Python/import.c

```diff
@@ -2249,6 +2249,9 @@ _imp_create_dynamic_impl(PyObject *module, PyObject *spec, PyObject *file)
 
     mod = _PyImport_FindExtensionObject(name, path);
     if (mod != NULL || PyErr_Occurred()) {
+        if (PyErr_Occurred()) {
+            PyErr_Clear();
```

Comment:
I've started a discussion over on the issue where if I'm right and swallowing an exception is bad then this clearing of the exception should be removed.